### ISSUE TITLE
Shellapi.h error. Unknown type REFIID

### ISSUE THREAD

**Issue by sotoea** [vlang/v#2435]
**V version:** 
0.1.21 b733876

**OS:** 
Windows 10

**What did you do?**
Tried running v.v and examples. I have installed V as well as Mingw with no major issues.

**What did you expect to see?**
The program compiles and runs with no issues.
 
**What did you see instead?**
```
PS D:\v> v .\v.v
In file included from D:\v\v.tmp.c:77:
d:\mingw\include\shellapi.h:332:35: error: unknown type name 'REFIID'; did you mean 'REFGUID'?
 HRESULT WINAPI SHGetImageList(int,REFIID,void**);
               ...
(Use `v -g` to print the entire error message)

V error: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```
-------After running with -g I got:--------------

```
PS D:\v> v -g .\v.v
C compiler=gcc
all .v files:
["D:\v\vlib\builtin\array.v", "D:\v\vlib\builtin\builtin.v", "D:\v\vlib\builtin\float.v", "D:\v\vlib\builtin\hashmap.v", "D:\v\vlib\builtin\int.v", "D:\v\vlib\builtin\map.v", "D:\v\vlib\builtin\option.v", "D:\v\vlib\builtin\string.v", "D:\v\vlib\builtin\utf8.v", "D:\v\vlib\strings\builder_c.v", "D:\v\vlib\strings\similarity.v", "D:\v\vlib\strings\strings.v", "D:\v\vlib\strconv\atoi.v", "D:\v\vlib\os\const.v", "D:\v\vlib\os\os.v", "D:\v\vlib\os\os_win.v", "D:\v\vlib\rand\pcg32.v", "D:\v\vlib\rand\rand.v", "D:\v\vlib\rand\splitmix64.v", "D:\v\vlib\term\can_show_color.v", "D:\v\vlib\term\colors.v", "D:\v\vlib\term\colors_win.v", "D:\v\vlib\term\control.v", "D:\v\vlib\time\time.v", "D:\v\vlib\vweb\tmpl\tmpl.v", "D:\v\vlib\benchmark\benchmark.v", "D:\v\vlib\compiler\cc.v", "D:\v\vlib\compiler\cflags.v", "D:\v\vlib\compiler\cgen.v", "D:\v\vlib\compiler\cheaders.v", "D:\v\vlib\compiler\compile_errors.v", "D:\v\vlib\compiler\comptime.v", "D:\v\vlib\compiler\depgraph.v", "D:\v\vlib\compiler\fn.v", "D:\v\vlib\compiler\gen_c.v", "D:\v\vlib\compiler\jsgen.v", "D:\v\vlib\compiler\live.v", "D:\v\vlib\compiler\main.v", "D:\v\vlib\compiler\module_header.v", "D:\v\vlib\compiler\modules.v", "D:\v\vlib\compiler\msvc.v", "D:\v\vlib\compiler\optimization.v", "D:\v\vlib\compiler\parser.v", "D:\v\vlib\compiler\parser2.v", "D:\v\vlib\compiler\query.v", "D:\v\vlib\compiler\repl.v", "D:\v\vlib\compiler\scanner.v", "D:\v\vlib\compiler\table.v", "D:\v\vlib\compiler\token.v", "D:\v\vlib\compiler\vfmt.v", "D:\v\vlib\compiler\vhelp.v", "D:\v\vlib\compiler\vtest.v", "v.v"]
In file included from D:\v\v.v:177:
d:\mingw\include\shellapi.h:332:35: error: unknown type name 'REFIID'; did you mean 'REFGUID'?
 HRESULT WINAPI SHGetImageList(int,REFIID,void**);
                                   ^~~~~~
                                   REFGUID
D:\v\vlib\builtin\builtin.v: In function 'builtin__init':
D:\v\vlib\builtin\builtin.v:10:34: error: '_O_U16TEXT' undeclared (first use in this function); did you mean 'O_TEXT'?
    C._setmode(C._fileno(C.stdin), C._O_U16TEXT)
                                  ^~~~~~~~~~
                                  O_TEXT
D:\v\vlib\builtin\builtin.v:10:34: note: each undeclared identifier is reported only once for each function it appears in
D:\v\vlib\builtin\builtin.v:12:34: error: '_O_U8TEXT' undeclared (first use in this function); did you mean 'O_TEXT'?
    C._setmode(C._fileno(C.stdin), C._O_U8TEXT)
                                  ^~~~~~~~~
                                  O_TEXT
In file included from D:\v\v.v:106:
D:\v\vlib\os\os_win.v: In function 'os__get_file_handle':
D:\v\vlib\os\os_win.v:117:33: error: request for member '_file' in something not a structure or union
     _handle := C._get_osfhandle(C._fileno(_fd)) // CreateFile? - hah, no -_-
                                 ^~~~~~~
V error: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

**Comment by sotoea** [comment]
I have changed _O_U8TEXT and _O_U16TEXT with 0_TEXT as suggested by the output message and the error disappears, not sure if this is a big issue that will break things down the line. 

For the shellapi, the error persists, any ideas?

**Comment by sotoea** [comment]
Ok, so, I am dumb, I had installed Mingw x32, started using Mingw x86_64 and errors went away. Now I have new issues but will open a new issue if I can't figure it out.

**Comment by Anabar** [comment]
I have the same problem in Windows 7 [Version 6.1.7601].
